source/non_matching/coupling.cc

```diff
@@ -40,6 +40,72 @@
 DEAL_II_NAMESPACE_OPEN
 namespace NonMatching
 {
+  namespace internal
+  {
+    template <int dim0, int dim1, int spacedim>
+    std::pair<std::vector<Point<spacedim>>, std::vector<unsigned int>>
+    compute_used_cells(const GridTools::Cache<dim0, spacedim> &cache,
+                       const DoFHandler<dim1, spacedim> &      immersed_dh,
```

Comment:
Is it possible that `dim0 != dim1`? It seems like you should be able to use just one variable for both.

Reply:
Well, good question!!
In principle one could have e.g. a 2d-disk inside a space which contains a 1d-curve and compute the coupling matrix, and I don't see why this code wouldn't work (but then, I also don't see why anyone would use a 3d space in such a situation). Anyhow, I just copy-pasted the original template choice, which was really made by @luca-heltai ...

Reply:
Yes, I believe that should be enforced: otherwise the "embedded manifold" would have greater dimension, which is clearly not possible...

Reply:
Then maybe put something like `static_assert (dim1 <= dim0, "...");` into the code.